crates/polars-core/src/frame/mod.rs

```diff
@@ -411,6 +411,29 @@ impl DataFrame {
         DataFrame { columns }
     }
 
+    /// Create a new `DataFrame` but does not check the length of the `Series`,
+    /// only check for duplicates.
+    ///
+    /// It is advised to use [Series::new](Series::new) in favor of this method.
+    ///
+    /// # Panic
+    /// It is the callers responsibility to uphold the contract of all `Series`
+    /// having an equal length, if not this may panic down the line.
+    pub fn new_no_length_checks(columns: Vec<Series>) -> PolarsResult<DataFrame> {
+        let mut names = PlHashSet::with_capacity(columns.len());
+        for column in &columns {
+            let name = column.name();
+            if !names.insert(name) {
+                // If insertion fails, it means the element is already in the set (duplicate)
+                polars_bail!(duplicate = name)
+            }
+        }
+        // we drop early as the brchk thinks the &str borrows are used when calling the drop
+        // of both `columns` and `names`
+        drop(names);
```

Comment:
taken from 

https://github.com/pola-rs/polars/blob/32df09e57818861e0758a307883b8602179bd18d/crates/polars-core/src/frame/mod.rs#L264-L266

I wouldn't have thought of this